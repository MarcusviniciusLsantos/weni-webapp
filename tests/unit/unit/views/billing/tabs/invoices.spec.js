import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';
import Invoices from '@/views/billing/tabs/invoices.vue';
import { vi } from 'vitest';

vi.mock('@/api/request.js', () => {});

describe('BillingInvoices.vue', () => {
  let wrapper;
  let store;
  let actions;

  beforeEach(() => {
    actions = {
      clearCurrentOrg: vi.fn(),
      clearCurrentProject: vi.fn(),
      createOrg: vi.fn(),
      changeAuthorization: vi.fn(),
      createProject: vi.fn(),
      setCurrentOrg: vi.fn(),
      setCurrentProject: vi.fn(),
      openModal: vi.fn(),
      setBillingOrgStep: vi.fn(),
      setBillingMembersStep: vi.fn(),
      setBillingProjectStep: vi.fn(),
      backBilling: vi.fn(),
    };

    store = createStore({
      actions,
    });

    wrapper = shallowMount(Invoices, {
      global: {
        plugins: [store],
        stubs: {
          RouterLink: RouterLinkStub,
          Indicator: true,
          UserManagement: true,
          Emoji: true,
          container: true,
          UnnnicButton: true,
          UnnnicSelect: true,
          UnnnicInput: true,
          UnnnicInputDatePicker: true,
          UnnnicTable: true,
          UnnnicTableRow: true,
          UnnnicSkeletonLoading: true,
          UnnnicCheckbox: true,
          UnnnicIconSvg: true,
        },
      },
    });
  });

  it('should be rendered properly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
