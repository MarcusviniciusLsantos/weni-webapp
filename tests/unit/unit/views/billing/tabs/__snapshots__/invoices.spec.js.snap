// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BillingInvoices.vue > should be rendered properly 1`] = `
<div
  data-v-98b068ca=""
>
  <div
    class="loading"
    data-v-98b068ca=""
    style="display: none;"
  >
    <div
      class="filters"
      data-v-98b068ca=""
    >
      <unnnicskeletonloading-stub
        data-v-98b068ca=""
        height="2.375rem"
        style="max-width: 18rem;"
        tag="div"
      />
    </div>
    <div
      class="table-header"
      data-v-98b068ca=""
    >
      <unnnicskeletonloading-stub
        data-v-98b068ca=""
        height="2.875rem"
        tag="div"
      />
    </div>
  </div>
  <div
    class="filters"
    data-v-98b068ca=""
  >
    <div
      class="date-picker-label"
      data-v-98b068ca=""
    >
       some specific text 
    </div>
    <unnnicinputdatepicker-stub
      actiontext="some specific text"
      cleartext="some specific text"
      data-v-98b068ca=""
      days="some specific text"
      inputformat="some specific text"
      months="some specific text"
      options="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
      size="sm"
      value="[object Object]"
    />
  </div>
  <unnnictable-stub
    class="invoices-table"
    data-v-98b068ca=""
    items=""
  >
    <unnnictablerow-stub
      data-v-98b068ca=""
      headers="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
    >
      <unnniccheckbox
        class="checkbox"
        data-v-98b068ca=""
      />
      <div
        data-v-98b068ca=""
        style="display: flex; align-items: center;"
      >
        <div
          class="break-text"
          data-v-98b068ca=""
          style="flex: 1;"
        >
           some specific text 
        </div>
        <unnniciconsvg
          clickable=""
          data-v-98b068ca=""
          icon="sort-default"
          scheme="neutral-clean"
          size="xs"
        />
      </div>
      <div
        data-v-98b068ca=""
        style="display: flex; align-items: center;"
      >
        <div
          class="break-text"
          data-v-98b068ca=""
          style="flex: 1;"
        >
           some specific text 
        </div>
        <unnniciconsvg
          clickable=""
          data-v-98b068ca=""
          icon="sort-default"
          scheme="neutral-clean"
          size="xs"
        />
      </div>
      <div
        data-v-98b068ca=""
        style="display: flex; align-items: center;"
      >
        <div
          class="break-text"
          data-v-98b068ca=""
          style="flex: 1;"
        >
           some specific text 
        </div>
        <unnniciconsvg
          clickable=""
          data-v-98b068ca=""
          icon="sort-default"
          scheme="neutral-clean"
          size="xs"
        />
      </div>
      <div
        class="action"
        data-v-98b068ca=""
      >
         some specific text 
      </div>
    </unnnictablerow-stub>
    <unnnictablerow-stub
      data-v-98b068ca=""
      headers="[object Object],[object Object],[object Object],[object Object],[object Object],[object Object]"
    />
  </unnnictable-stub>
  <infiniteloading-stub
    data-v-98b068ca=""
    empty="true"
    hideerrorslot="true"
  >
    <div
      class="loading"
      data-v-98b068ca=""
    >
      <div
        class="table-row"
        data-v-98b068ca=""
      >
        <unnnicskeletonloading-stub
          data-v-98b068ca=""
          height="2.25rem"
          tag="div"
        />
      </div>
      <div
        class="table-row"
        data-v-98b068ca=""
      >
        <unnnicskeletonloading-stub
          data-v-98b068ca=""
          height="2.25rem"
          tag="div"
        />
      </div>
      <div
        class="table-row"
        data-v-98b068ca=""
      >
        <unnnicskeletonloading-stub
          data-v-98b068ca=""
          height="2.25rem"
          tag="div"
        />
      </div>
      <div
        class="table-row"
        data-v-98b068ca=""
      >
        <unnnicskeletonloading-stub
          data-v-98b068ca=""
          height="2.25rem"
          tag="div"
        />
      </div>
    </div>
  </infiniteloading-stub>
</div>
`;
